cmd/csaf_provider/create.go

```diff
@@ -153,10 +154,55 @@ func createFeedFolders(c *config, wellknown string) error {
 				return err
 			}
 		}
+		// Create an empty ROLIE feed document
+		if err := createROLIEfeed(c, t, tlpLink); err != nil {
+			return err
+		}
 	}
 	return nil
 }
 
+// createROLIEfeed creates an empty ROLIE feed
+func createROLIEfeed(c *config, t tlp, folder string) error {
+	ts := string(t)
+	feedName := "csaf-feed-tlp-" + ts + ".json"
+
+	feed := filepath.Join(folder, feedName)
+
+	feedURL := csaf.JSONURL(
+		c.CanonicalURLPrefix +
+			"/.well-known/csaf/" + ts + "/" + feedName)
+
+	tlpLabel := csaf.TLPLabel(strings.ToUpper(ts))
+
+	links := []csaf.Link{{
+		Rel:  "self",
+		HRef: string(feedURL),
+	}}
+	// If we have a service document we need to link it.
+	if c.ServiceDocument {
+		links = append(links, csaf.Link{
+			Rel:  "service",
+			HRef: c.CanonicalURLPrefix + "/.well-known/csaf/service.json",
+		})
+	}
+	rolie := &csaf.ROLIEFeed{
+		Feed: csaf.FeedData{
+			ID:    "csaf-feed-tlp-" + ts,
+			Title: "CSAF feed (TLP:" + string(tlpLabel) + ")",
+			Link:  links,
+			Category: []csaf.ROLIECategory{{
+				Scheme: "urn:ietf:params:rolie:category:information-type",
+				Term:   "csaf",
+			}},
+			Updated: csaf.TimeStamp(time.Now().UTC()),
```

Comment:
The `entry` array should be added here as empty array: `Entry: []*csaf.Entry{},`

Reply:
Addressed in c4950065475d4314efb5f171cfad16435940bed2

Reply:
commit 26a4d2e7b94f0564f7d3a27ee419fb004e888392 removes the `omitempty` rule which eliminates empty elements from the JSON result. This should finally help.

Reply:
This is resolved.